components/operations-controller/controllers/operation_controller_integration_test.go

```diff
@@ -0,0 +1,692 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers_test
+
+import (
+	"context"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	web_hook "github.com/kyma-incubator/compass/components/director/pkg/webhook"
+	"github.com/kyma-incubator/compass/components/operations-controller/api/v1alpha1"
+	"github.com/kyma-incubator/compass/components/operations-controller/controllers"
+	"github.com/kyma-incubator/compass/components/operations-controller/controllers/controllersfakes"
+	"github.com/kyma-incubator/compass/components/operations-controller/internal/k8s"
+	"github.com/kyma-incubator/compass/components/operations-controller/internal/k8s/status"
+	"github.com/kyma-incubator/compass/components/operations-controller/internal/webhook"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/director"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	kubeerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"path/filepath"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"testing"
+)
+
+const (
+	eventuallyTimeout = 60 * time.Second
+	eventuallyTick    = 1 * time.Second
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	err := clientgoscheme.AddToScheme(scheme)
+	if err != nil {
+		panic(err)
+	}
+
+	err = v1alpha1.AddToScheme(scheme)
+	if err != nil {
+		panic(err)
+	}
+
+}
+
+/*
+
+ The following tests present semi-integration style tests where a lightweight
+ in-memory Kubernetes API Server is deployed in order to test out some of the critical flows
+ of the Operation Controller reconciliation loop.
+
+ Unlike the tests in operation_controller_test.go the tests here do not aim to be as exhaustive as possible
+ but rather strive to cover main flows and scenarios. Also - similarly to the operation_controller_test.go
+ tests, here the Director and Webhook clients have been mocked.
+
+*/
+
+func TestController_Scenarios(t *testing.T) {
```

Comment:
you could verify that deleting an op does not trigger the controller similarly to how you did for status update (separate test or one of the existing tests?)